tests/phpunit/Elastic/Indexer/Attachment/ScopeMemoryLimiterTest.php

```diff
@@ -89,7 +89,7 @@ public function testExecute() {
 		$converter = new ScopeMemoryLimiter();
 
 		if ( $memoryLimitBefore === "-1" ) {
-			$memoryLimitBefore = memory_get_usage() + $converter->toInt( '10M' );
+			$memoryLimitBefore = memory_get_usage() + $converter->toInt( '20M' );
```

Comment:
A fixed value like '20M' doesn’t consider the current memory usage or the system's overall memory limit. If the current memory usage already exceeds the fixed value, the test will fail. For more complex or memory-intensive tests, 20 MB might be insufficient, while in some cases, it's unnecessarily large.

@JeroenDeDauw what do you think? I'd suggest to follow a more dynamic solution that should:

- Take the current memory usage into account.
- Ensure that the new memory limit is always above the current memory usage.
- Provide a configurable buffer to account for additional memory needs.

Reply:
@gesinn-it-gea, @JeroenDeDauw maybe this approach or something like this:

```
public function testExecute() {
		// Store the original memory limit
		$originalMemoryLimitBefore = ini_get( 'memory_limit' );
		$converter = new ScopeMemoryLimiter();
	
		// Get current memory usage
		$currentMemoryUsage = memory_get_usage();
	
		// Set a configurable buffer (e.g., 20MB) to ensure the new memory limit is above current usage
		$buffer = $converter->toInt( '20M' );
		
		// Calculate new memory limit (current usage + buffer)
		$memoryLimitBefore = $currentMemoryUsage + $buffer;
	
		// If the original memory limit is set to -1 (unlimited), set it to the calculated value
		if ( $originalMemoryLimitBefore === "-1" ) {
			ini_set( 'memory_limit', $memoryLimitBefore );
		}
	
		// Mock a callable to test the execution
		$this->testCaller = $this->getMockBuilder( '\stdClass' )
			->disableOriginalConstructor()
			->setMethods( ['calledFromCallable'] )
			->getMock();
	
		$this->testCaller->expects( $this->once() )
			->method( 'calledFromCallable' );
	
		// Calculate the final target memory limit (including a small increment)
		$memoryLimit = $memoryLimitBefore + $converter->toInt( '1M' );
	
		// Create an instance of ScopeMemoryLimiter with the new memory limit
		$instance = new ScopeMemoryLimiter( $memoryLimit );
	
		$instance->execute( [ $this, 'runCallable' ] );
	
		$this->assertEquals(
			$memoryLimit,
			$this->memoryLimitFromCallable,
			"Limit we expected got set."
		);
	
		$this->assertEquals(
			$memoryLimitBefore,
			$instance->getMemoryLimit(),
			"Limit was reset successfully."
		);
	
		ini_set( 'memory_limit', $originalMemoryLimitBefore );
	}
```